### DISCUSSION TITLE
Sort and Compare Locale keys

### DISCUSSION THREAD

**Discussion by timheerwagen** [vinissimus/next-translate#882] [Q&A]
I am currently still looking for a good solution to sort language files and compare different languages for missing keys.

The only solution I found was currently only this eslint plugin: https://github.com/godaddy/eslint-plugin-i18n-json

This unfortunately only works partially, has a vulnerable peer dependency and also isn't maintained anymore.

**Comment by aralroca** [comment]
You can sort them creating an script and running then this script in your pre-commit:

```js
const fs = require('fs')
const config = require('i18n.js')
const sortJson = require('sort-json')

Object.keys(config.locales).forEach((lang) => {
  const dir = `src/locales/${lang}`
  const files = fs.readdirSync(dir)
  files.forEach((file) => {
    sortJson.overwrite(`${dir}/${file}`)
  })
})
```

To log the missing keys you can use the [`logger`](https://github.com/vinissimus/next-translate#3-configuration) property of the `i18n.js` configuration
